Add OpenHands Cloud API documentation (#8127)
All-Hands-AI/OpenHands: docs/modules/usage/cloud/cloud-api.md

This guide explains how to obtain an API key and use the API to start conversations.
For more detailed information about the API, refer to the [OpenHands API Reference](https://docs.all-hands.dev/swagger-ui/).

## Obtaining an API Key

To use the OpenHands Cloud API, you'll need to generate an API key:

1. Log in to your [OpenHands Cloud](https://app.all-hands.dev) account
2. Navigate to the [Settings page](https://app.all-hands.dev/settings) by clicking on your profile icon in the bottom-left corner
3. Locate the "API Keys" section
4. Click "Generate New Key"
5. Give your key a descriptive name (e.g., "Development", "Production")
6. Copy the generated API key and store it securely - it will only be shown once

![API Key Generation](/img/docs/api-key-generation.png)

## API Usage

### Starting a New Conversation

To start a new conversation with OpenHands performing a task, you'll need to make a POST request to the conversation endpoint.

#### Request Parameters

| Parameter | Type | Required | Description |
|-----------|------|----------|-------------|
| `initial_user_msg` | string | Yes | The initial message to start the conversation |

> | `repository` | string | No | GitHub repository URL to provide context (must be a repository you've granted access to) |

Reviewer: We should give an example for the repo param perhaps. This has to be `owner/repo` format, rather than URL links to the repo

#### Examples

<details>
<summary>cURL</summary>

```bash
curl -X POST "https://app.all-hands.dev/api/conversations" \
  -H "Authorization: Bearer YOUR_API_KEY" \
  -H "Content-Type: application/json" \
  -d '{
    "initial_user_msg": "Check whether there is any incorrect information in the README.md file and send a PR to fix it if so.",

>     "repository": "https://github.com/yourusername/your-repo"

Reviewer: ```suggestion
    "repository": "yourusername/your-repo"
```

Ah I see the CURL example is here; no host information needed

    console.log(`Conversation Link: https://app.all-hands.dev/conversations/${conversation.id}`);
    console.log(`Status: ${conversation.status}`);

    return conversation;
  } catch (error) {
    console.error("Error starting conversation:", error);
  }
}

startConversation();
```

</details>

#### Response

The API will return a JSON object with details about the created conversation:

```json
{
  "status": "ok",
  "conversation_id": "abc1234",
}
```

### Retrieving Conversation Status

You can check the status of a conversation by making a GET request to the conversation endpoint.

#### Endpoint

```
GET https://app.all-hands.dev/api/conversations/{conversation_id}
```

#### Example

<details>
<summary>cURL</summary>

```bash
curl -X GET "https://app.all-hands.dev/api/conversations/{conversation_id}" \
  -H "Authorization: Bearer YOUR_API_KEY"
```
</details>

#### Response

```json
{
  "conversation_id":"abc1234",
  "title":"Update README.md",
  "created_at":"2025-04-29T15:13:51.370706Z",
  "last_updated_at":"2025-04-29T15:13:57.199210Z",
  "status":"RUNNING",
  "selected_repository":"https://github.com/yourusername/your-repo",
  "trigger":"gui"
}
```

> ## Rate Limits

Reviewer: Similarly having a section for `Authentication error` would be helpful

1. Either you provided the wrong repo name
2. Or you don't have access to the repo

This guide explains how to obtain an API key and use the API to start conversations.
For more detailed information about the API, refer to the [OpenHands API Reference](https://docs.all-hands.dev/swagger-ui/).

## Obtaining an API Key

To use the OpenHands Cloud API, you'll need to generate an API key:

1. Log in to your [OpenHands Cloud](https://app.all-hands.dev) account
2. Navigate to the [Settings page](https://app.all-hands.dev/settings)
3. Locate the "API Keys" section
4. Click "Generate New Key"
5. Give your key a descriptive name (e.g., "Development", "Production")
6. Copy the generated API key and store it securely - it will only be shown once

![API Key Generation](/img/docs/api-key-generation.png)

## API Usage

### Starting a New Conversation

To start a new conversation with OpenHands performing a task, you'll need to make a POST request to the conversation endpoint.

#### Request Parameters

| Parameter | Type | Required | Description |
|-----------|------|----------|-------------|
| `initial_user_msg` | string | Yes | The initial message to start the conversation |

> | `repository` | string | No | GitHub repository name to provide context in the format `owner/repo`. You must have access to the repo. |

Reviewer: ```suggestion
| `repository` | string | No | Git repository name to provide context in the format `owner/repo`. You must have access to the repo. |
```

Can be either Gitlab or Github repo (depends on whether use has logged in with the git provider on saas)

# OpenHands Cloud API

> OpenHands Cloud provides a REST API that allows you to programmatically interact with the service. This guide explains how to obtain an API key and use the API to start conversations.

Author: ```suggestion
OpenHands Cloud provides a REST API that allows you to programmatically interact with the service. This is useful if you easily want to kick off your own jobs from your programs in a flexible way.

This guide explains how to obtain an API key and use the API to start conversations.
```

## Obtaining an API Key

To use the OpenHands Cloud API, you'll need to generate an API key:

> 1. Log in to your OpenHands Cloud account

Author: ```suggestion
1. Log in to your [OpenHands Cloud](https://app.all-hands.dev) account
```

> 2. Navigate to the Settings page by clicking on your profile icon in the top-right corner

Author: ```suggestion
2. Navigate to the Settings page by clicking on your profile icon in the bottom-left corner
```
